### PR TITLE
Tilde expanding file paths

### PR DESCRIPTION
**Pull request by UseAfterFree-177**
As a Unix/Linux guy I expect that `~/` expands to my home directory (`/mnt/usr/u/home/use-after-free/`), and that I can have my gamesaves in `~/.saves/WM` and girl files in `~/WM-girls`.

### CODE REVIEW

File: src/engine/utils/DirPath.cpp [hunk 1]
```diff
@@ -65,3 +70,60 @@ ButtonPath::ButtonPath(const std::string& filename)
...
+
+  // Expand "~/some/dir" to "<my home-dir>/some/dir"
+  //
+  // TODO: Expand "~mike/dir" to "<mike's home-dir>/dir"
+  std::string tildeexpand(std::string path)
+  {
+    if(path.empty())
+      return path;
+
+    if(path.substr(0, 2) == "~/")
```
Comment (h1262216) [inline src/engine/utils/DirPath.cpp:113]: could you please use braces if the body of the if spans multiple lines? I think that makes it less error prone. Also, maybe indent the body to make things more readable.

**Comment by h1262216** [comment]
nice addition. Do we really need both versions of getting the user home? passwd and env variable? 
do we need the additional indirection with the unique_ptr in pwd? would we consider the empty string to be a valid home directory?

**Comment by UseAfterFree-177** [comment]
1. $HOME is the friendly way, you, the user, can easily change it whenever  you want to. passwd is the way that (almost) always work, but only your friendly neighbourhood sysadmin can change it.
2. I'll see if there's some kind of optional<> I can use. Otherwise I go with the empty string.

**Comment by UseAfterFree-177** [comment]
Done. I've replaced `std::unique_ptr<>` with `boost::optional<>`.

File: src/engine/utils/DirPath.cpp (hunk 1 above)
Comment (UseAfterFree-177) [reply src/engine/utils/DirPath.cpp:113]: Fixed. And thanks for pointing it out.

Comment (UseAfterFree-177) [reply src/engine/utils/DirPath.cpp:113]: Huh, you fixed it too?

Oh, well... :+1:

**Pull request merged**